ganga/GangaCore/Lib/BackendScriptTemplate.py.template

```diff
@@ -0,0 +1,207 @@
+    errfile.close()
+    print('EXITCODE: %d' % -9999, file=filehandle)
+    print('FAILED: %s'%time.strftime('%a %b %d %H:%M:%S %Y'), file=filehandle)
+    print('PROBLEM STARTING THE APPLICATION SCRIPT: \'%s\' \'%s\''%(execmd,str(x)), file=filehandle)
+    print('FILES FOUND ARE: %s' % os.listdir('.'), file=filehandle)
+    filehandle.close()
+    sys.exit()
+
+statusfilename = os.path.join(sharedoutputpath,'__jobstatus__')
+statusfile=open_file(statusfilename)
+
+if Backend == "BATCH":
+    heartbeatfilename = os.path.join(sharedoutputpath,'__heartbeat__')
+    heartbeatfile=open_file(heartbeatfilename)
+
+
+line='START: '+ time.strftime('%a %b %d %H:%M:%S %Y',time.gmtime(time.time())) + os.linesep
+if Backend == "BATCH":
+    try:
+        line+='PID: ' + os.getenv('###JOBIDNAME###') + os.linesep
+        line+='QUEUE: ' + os.getenv('###QUEUENAME###') + os.linesep
+        line+='ACTUALCE: ' + hostname() + os.linesep
+    except:
+        pass
+
+statusfile.writelines(line)
+flush_file(statusfile)
+
+if Backend == "LOCAL":
+    workdir = ###WORKDIR###
+    if not os.path.exists(workdir):
+        os.makedirs(workdir)
+    os.chdir(workdir)
+
+
+for f in input_sandbox:
+    if mimetypes.guess_type(f)[1] in ['gzip', 'bzip2']:
+        getPackedInputSandbox(f)
+    else:
+        shutil.copy(f, os.path.join(os.getcwd(), os.path.basename(f)))
+
+# -- END OF MOVED CODE BLOCK
+
+
+#get input files
+###DOWNLOADINPUTFILES###
+
+# create inputdata list
+###CREATEINPUTDATALIST###
+
+gangadir = ###GANGADIR###
+sys.path.insert(0,gangadir)
+sys.path.insert(0,os.path.join(os.getcwd(),PYTHON_DIR))
+
+runenv = os.environ.copy()
+for key,value in environment.items():
+    runenv[key] = value
+
+if Backend == "BATCH":
+    outfile = os.dup(sys.stdout.fileno())
+    errorfile = os.dup(sys.stderr.fileno())
+
+    print("--- GANGA APPLICATION OUTPUT BEGIN ---", file=sys.stdout)
+    print("--- GANGA APPLICATION ERROR BEGIN ---", file=sys.stderr)
+    flush_file(sys.stdout)
+    flush_file(sys.stderr)
+
+elif Backend == "LOCAL":
+    outfile=open('stdout','w')
+    errorfile=open('stderr','w')
+
+sys.stdout=open('./__syslog__','w')
+sys.stderr=sys.stdout
+
+###VIRTUALIZATION###
+
+try:
+    child = subprocess.Popen(execmd, shell=False, stdout=outfile, stderr=errorfile, env=runenv)
+except OSError as x:
+    failurereport(statusfile, 'PROBLEM STARTING THE APPLICATION SCRIPT: \'%s\' \'%s\''%(execmd,str(x)))
+
+if Backend == "LOCAL":
+    print('PID: %d'%child.pid, file=statusfile)
+    flush_file(statusfile)
+
+result = -1
+
+try:
+    while 1:
+        result = child.poll()
+        if result is not None:
+            break
+        if Backend=="LOCAL":
+            flush_file(outfile)
+            flush_file(errorfile)
+            time.sleep(0.3)
+        elif Backend=="BATCH":
+            heartbeatfile.write('.')
+            flush_file(heartbeatfile)
+            heartbeatsequence = ###HEARTBEATFREQUENCE###
+            time.sleep(int(heartbeatsequence))
```

Comment:
What about changing the monitoring of Local jobs to use the hearbeatfile as well. In that way we avoid some of thse if/else statements and make code more uniform.